packages/store/src/transformer/job.ts

```diff
@@ -196,8 +196,18 @@ export class Job {
       index
     );
 
-    for (const [index, child] of children.entries()) {
-      await this._snapshotToBlock(child, page, id, index);
+    // surface-block needs to be initialized at last.
+    // because the reference id in group and connector need to be updated.
+    if (snapshot.flavour === 'affine:page') {
```

Comment:
We should avoid bringing in special treatment about `affine:xxx` in store logic. Other workarounds? cc @Saul-Mirone